File: Python/String-2/count_hi.py
"""
Dada uma String "str", retorne o número de vezes em que a string "hi" aparece.
"""


def count_hi(str):
    return str.count("hi")


# OUTRO JEITO MENOS DIRETO E COM LOOPING
def count_hi_2(str):
    cont = 0
    for i in range(len(str)):
        if str[i:i+2] == "hi":
            cont += 1
    return cont


print(count_hi("hi World"))
print(count_hi_2("hi World"))

File: Java/String-2/src/countHi/CountHi.java
package countHi;

public class CountHi {
    //Retorne a quantidade de vezes que a palavra "hi" aparece na String dada.

    public int countHi(String str) {
        int cont = 0;
        for (int i = 0; i < str.length()-1; i++){
            if (str.substring(i,i+2).equals("hi")) cont++;
        }
        return cont;
    }
}
